### PR TITLE
Refactor sub ctx and modify all relative code

### PR DESCRIPTION
**Pull request by wanghaEMQ**

### CODE REVIEW

File: nanomq/include/broker.h [hunk 1]
```diff
@@ -54,9 +54,10 @@ struct client_ctx {
 #ifdef STATISTICS
 	nng_atomic_u64 *recv_cnt;
 #endif
-	conn_param *             cparam;
-	struct packet_subscribe *sub_pkt;
-	uint8_t                  proto_ver;
+	conn_param *cparam;
+	uint32_t    prop_len;
```
Comment (JaylinYu) [inline nanomq/include/broker.h:58]: could you get rid of prop from cli_ctx?

File: nanomq/pub_handler.c [hunk 2]
```diff
@@ -126,9 +126,10 @@ foreach_client(
 		if (0 == dbtree_ctxt_free(db_ctxt)) {
 			client_ctx *ctx = dbtree_ctxt_delete(db_ctxt);
 			if (ctx) {
-				// Bug here
-				// del_sub_ctx(ctx, topic);
-				destroy_sub_client(ctx->pid.id, pub_work->db);
+				if (!dbhash_check_id(ctx->pid.id))
+					sub_ctx_free(ctx);
+				// sub_ctx_del(db, topic, pid);
+				// destroy_sub_client(ctx->pid.id, pub_work->db);
```
Comment (JaylinYu) [inline nanomq/pub_handler.c:132]: plz delete these code

**Review by JaylinYu — APPROVED** [review]
is there anything more you need to change?

File: nanomq/include/broker.h (hunk 1 above)
Comment (wanghaEMQ) [reply nanomq/include/broker.h:58]: I am afraid not, user property in prop should keep available in the broker.c

**Review by JaylinYu — APPROVED** [review]

**Pull request merged**